rs/sns/testing/src/sns.rs

```diff
@@ -198,40 +280,162 @@ async fn complete_sns_swap<C: CallCanisters + ProgressNetwork>(
                 .await
-                .expect("Failed to get auto finalization status");
+                .map_err(|e| format!("Failed to get auto finalization status: {e}"))?;
             is_auto_finalization_status_committed_or_err(&auto_finalization_status)
         },
         &Ok(true),
     )
-    .await
-    .unwrap();
+    .await?;
+
+    let sns_nervous_system_parameters = governance_canister
+        .get_nervous_system_parameters(agent)
+        .await
+        .map_err(|e| format!("Failed to get nervous system parameters: {e}"))?;
+    let neuron_mininum_disolve_delay = sns_nervous_system_parameters
+        .neuron_minimum_dissolve_delay_to_vote_seconds
+        .ok_or("Expecting the neuron minimum dissolve delay to be set")?
+        as u32;
+
+    println!(
+        "Increasing dissolve delay to {} for swap participants...",
+        neuron_mininum_disolve_delay
+    );
+    for swap_participant_agent in swap_participants {
+        let swap_participant_neuron_id =
+            get_caller_neuron(&swap_participant_agent, governance_canister)
+                .await
+                .map_err(|e| {
+                    format!(
+                        "Failed to get the caller neuron for {}: {e}",
+                        swap_participant_agent.caller().unwrap()
+                    )
+                })?;
+        if let Some(swap_participant_neuron_id) = swap_participant_neuron_id {
+            for neuron_to_follow in &neurons_to_follow {
+                let follow = Follow {
+                    followees: vec![neuron_to_follow.clone()],
+                    // UpgradeSnsControlledCanister
+                    // TODO: @rvem: Do we need to enable other functions following too?
+                    function_id: 3,
+                };
+                governance_canister
+                    .follow(
+                        &swap_participant_agent,
+                        swap_participant_neuron_id.clone(),
+                        follow,
+                    )
+                    .await
+                    .map_err(|e| {
+                        format!(
+                            "Neuron {:?} failed to follow the neuron: {}",
+                            swap_participant_neuron_id.clone(),
+                            e
+                        )
+                    })?;
+            }
+
+            governance_canister
+                .increase_dissolve_delay(
+                    &swap_participant_agent,
+                    swap_participant_neuron_id.clone(),
+                    neuron_mininum_disolve_delay,
+                )
+                .await
+                .map_err(|e| {
+                    format!(
+                        "Failed to increase dissolve delay for neuron {:?}: {}",
+                        swap_participant_neuron_id, e
+                    )
+                })?;
+        }
+    }
+
+    Ok(())
+}
+
+pub async fn sns_proposal_upvote<
```

Comment:
This function would be nice to have in the nervous system agent, as it could then be used in producton scenarios. 

Low-prio request

Reply:
Okay, I now see this is more specific to SNS testing. Nevermind